daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -156,6 +156,45 @@ import org.apache.daffodil.xml.QName
 import org.apache.daffodil.xml.XMLUtils
 import org.apache.daffodil.grammar.primitives.ElementCombinator
 import org.apache.daffodil.grammar.primitives.TypeValueCalc
+import org.apache.daffodil.grammar.primitives.LogicalNilValue
+import org.apache.daffodil.grammar.primitives.ConvertZonedIntPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextShortPrim
+import org.apache.daffodil.grammar.primitives.TrailingSkipRegion
+import org.apache.daffodil.grammar.primitives.ConvertZonedUnsignedBytePrim
+import org.apache.daffodil.grammar.primitives.DynamicEscapeSchemeCombinatorElement
+import org.apache.daffodil.grammar.primitives.ConvertTextDecimalPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextBooleanPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextBytePrim
+import org.apache.daffodil.grammar.primitives.ConvertZonedBytePrim
+import org.apache.daffodil.grammar.primitives.ConvertZonedUnsignedShortPrim
+import org.apache.daffodil.grammar.primitives.AlignmentFill
+import org.apache.daffodil.grammar.primitives.ComplexNilOrContent
+import org.apache.daffodil.grammar.primitives.LeadingSkipRegion
+import org.apache.daffodil.grammar.primitives.ConvertTextFloatPrim
+import org.apache.daffodil.grammar.primitives.ConvertZonedIntegerPrim
+import org.apache.daffodil.dsom.ExpressionCompilers
+import org.apache.daffodil.grammar.primitives.ConvertZonedCombinator
+import org.apache.daffodil.grammar.primitives.ConvertZonedUnsignedLongPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextUnsignedIntPrim
+import org.apache.daffodil.grammar.primitives.ConvertZonedShortPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextUnsignedBytePrim
+import org.apache.daffodil.grammar.primitives.ConvertZonedDecimalPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextUnsignedLongPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextIntPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextLongPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextNonNegativeIntegerPrim
+import org.apache.daffodil.grammar.primitives.ConvertZonedUnsignedIntPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextUnsignedShortPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextIntegerPrim
+import org.apache.daffodil.grammar.primitives.ConvertTextDoublePrim
+import org.apache.daffodil.grammar.primitives.LiteralCharacterNilOfSpecifiedLength
+import org.apache.daffodil.grammar.primitives.DelimiterStackCombinatorElement
+import org.apache.daffodil.grammar.primitives.ConvertTextCombinator
+import org.apache.daffodil.runtime1.ElementBaseRuntime1Mixin
+import org.apache.daffodil.grammar.primitives.ConvertZonedNonNegativeIntegerPrim
```

Comment:
Are all these new import statements needed (I'm guessing they may be for #289 ).  Also, just for my own knowledge, I'm assuming we are avoiding doing an `import org.apache.daffodil.grammar.primitives._` in order to make it clear where things are coming from, is that correct?

Reply:
I don't fine it helpful that these are enumerated. Our standard is to do them individually, but I am not sure why. In this case, it just makes the code harder to work on. If you import primitives._ then code-completion works and you get the symbols you want without having to remember their specific spellings or search around. 

I don't think the import statements are useful for isolating dependencies or anything like that.  

I will change this to just use a wildcard import for grammar.primitives._